### PR TITLE
Replace SPM dependency with Swift Tool Support

### PR DESCRIPTION
**Pull request by lakpa**
Signed-off-by: ldindu <lakpadindu@hotmail.com>

*Issue number of the reported bug or feature request: #46*

**Describe your changes**
Replacing **Swift Package Manager** dependency with **Swift Tool Support** to reduce the dependency footprint.

**Testing performed**
Not applicable

### CODE REVIEW

**Comment by marciniwanicki** [comment]
Many thanks @lakpa for the PR. Looks very good!

The CI failure is probably not related to your changes. I will investigate it this week. 

```sh
xcrun swift test --enable-code-coverage
646[283/283] Linking xcdiffPackageTests
6472020-06-13 00:45:39.895 xctest[4465:9761] The bundle “xcdiffPackageTests.xctest” couldn’t be loaded because it is damaged or missing necessary resources. Try reinstalling the bundle.
6482020-06-13 00:45:39.895 xctest[4465:9761] (dlopen_preflight(/Users/travis/build/bloomberg/xcdiff/.build/x86_64-apple-macosx/debug/xcdiffPackageTests.xctest/Contents/MacOS/xcdiffPackageTests): Library not loaded: /System/Library/Frameworks/CryptoKit.framework/Versions/A/CryptoKit
649  Referenced from: /Users/travis/build/bloomberg/xcdiff/.build/x86_64-apple-macosx/debug/libSwiftToolsSupport.dylib
650  Reason: image not found)
651error: Error Domain=NSCocoaErrorDomain Code=260 "The folder “codecov” doesn’t exist." UserInfo={NSFilePath=/Users/travis/build/bloomberg/xcdiff/.build/x86_64-apple-macosx/debug/codecov, NSUserStringVariant=(
652    Folder
653), NSUnderlyingError=0x7fa50b564050 {Error Domain=NSOSStatusErrorDomain Code=-43 "fnfErr: File not found"}}
654make: *** [test] Error 1
655error: .build/debug/xcdiff: Failed to load coverage: No such file or directory
656error: Could not load coverage information
657The command "make clean test ; ./Scripts/coverage.sh" exited with 1
```

**Comment by marciniwanicki** [comment]
Thanks @lakpa, would you mind switching dependency to use `SwiftToolsSupport-auto` instead of `SwiftToolsSupport`? 

Looks that Swift Tools Support (https://github.com/apple/swift-tools-support-core/blob/master/Package.swift) has 2 variations: dynamic `SwiftToolsSupport` and static `SwiftToolsSupport-auto`. To ease the installation process we'd probably prefer to use static. I wonder if that would help with the CI failure.

We can also try to point to the latest `0.1.5` version.

**Comment by lakpa** [comment]
> https://github.com/apple/swift-tools-support-core/blob/master/Package.swift

Sure, I will get it updated

**Comment by lakpa** [comment]
> Thanks @lakpa, would you mind switching dependency to use `SwiftToolsSupport-auto` instead of `SwiftToolsSupport`?
> 
> Looks that Swift Tools Support (https://github.com/apple/swift-tools-support-core/blob/master/Package.swift) has 2 variations: dynamic `SwiftToolsSupport` and static `SwiftToolsSupport-auto`. To ease the installation process we'd probably prefer to use static. I wonder if that would help with the CI failure.
> 
> We can also try to point to the latest `0.1.5` version.

Sure, I will get it updated

**Comment by lakpa** [comment]
it appears to be complaining as it couldn't **CryptoKit.framework**

Screenshot 2020-06-21 at 21 36 08

File: Package.swift [hunk 1]
```diff
@@ -23,7 +23,7 @@ let package = Package(
         ),
     ],
     dependencies: [
-        .package(url: "https://github.com/apple/swift-package-manager", .upToNextMajor(from: "0.5.0")),
+        .package(url: "https://github.com/apple/swift-tools-support-core.git", .upToNextMajor(from: "0.1.0")),
```
Comment (kwridan) [inline Package.swift:26]: I believe if we bump the version to `0.1.5` we'll get the fix for this :)

https://github.com/apple/swift-tools-support-core/commit/f34f419c73d39c830a006ff4c113449801f1fb4f#diff-8a8089dc1f441b3f42878901d077d13cR11

Thanks @lakpa !

File: Package.resolved [hunk 2]
```diff
@@ -29,21 +29,12 @@
...
-      },
-      {
-        "package": "SwiftPM",
-        "repositoryURL": "https://github.com/apple/swift-package-manager",
-        "state": {
-          "branch": null,
-          "revision": "9abcc2260438177cecd7cf5185b144d13e74122b",
-          "version": "0.5.0"
+          "revision": "81e2cda46447b4d78c18e03b0d284d76950f7c2b",
+          "version": "0.1.5"
```
Comment (kwridan) [inline Package.resolved:37]: ah nvm it's already picked that version up - how odd :/

**Comment by lakpa** [comment]
Recent Xcode update change has fixed the failing CI issue, thanks @kwridan !!!

**Pull request merged**